src/main/java/uk/gov/hmcts/reform/cpo/repository/CasePaymentOrdersRepositoryImpl.java

```diff
@@ -45,11 +44,30 @@ public void deleteByCaseIds(List<Long> caseIds) {
     }
 
     private void validateAllEntriesExistByCaseIds(List<Long> caseIds) {
+        List<String> nonExistentCaseIds = new ArrayList<>();
         for (Long cid : caseIds) {
             if (casePaymentOrdersJpaRepository.findAllByCaseId(cid).isEmpty()) {
```

Comment:
Can we expose a count query instead of fetching whole entities as we are not using them anyway apart from validation?

Reply:
Fixed in https://github.com/hmcts/cpo-case-payment-orders-api/pull/82/commits/7503751b934686d8ada84a8f2ab06bcb589e7bf5